pandas/core/internals/blocks.py

```diff
@@ -1485,39 +1486,29 @@ def fillna(
     ) -> list[Block]:
         # Caller is responsible for validating limit; if int it is strictly positive
 
-        try:
-            new_values = self.values.fillna(value=value, limit=limit)
-        except (TypeError, ValueError) as err:
-            _catch_deprecated_value_error(err)
-
-            if is_interval_dtype(self.dtype):
-                # Discussion about what we want to support in the general
-                #  case GH#39584
-                blk = self.coerce_to_target_dtype(value)
-                return blk.fillna(value, limit, inplace, downcast)
-
-            elif isinstance(self, NDArrayBackedExtensionBlock):
-                # We support filling a DatetimeTZ with a `value` whose timezone
-                #  is different by coercing to object.
-                if self.dtype.kind == "m":
-                    # GH#45746
-                    warnings.warn(
-                        "The behavior of fillna with timedelta64[ns] dtype and "
-                        f"an incompatible value ({type(value)}) is deprecated. "
-                        "In a future version, this will cast to a common dtype "
-                        "(usually object) instead of raising, matching the "
-                        "behavior of other dtypes.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
-                    )
-                    raise
-                blk = self.coerce_to_target_dtype(value)
-                return blk.fillna(value, limit, inplace, downcast)
-
-            else:
+        if self.dtype.kind == "m":
+            try:
+                res_values = self.values.fillna(value, limit=limit)
```

Comment:
What's the reason for putting this behind a `if self.dtype.kind == "m":` check?

Because this now will call `super().fillna()` for all EAs, but the parent `fillna()` implementation doesn't actually call `EA.fillna()`

(this was catched in the geopandas test suite)

Reply:
Which future warning?

Let me rephrase my question: what's the reason for no longer calling EA.fillna() for all EAs?

Reply:
> Which future warning?

The FutureWarning on L1494-1502 just below this.

> Let me rephrase my question: what's the reason for no longer calling EA.fillna() here for all EAs?

Primarily code-sharing, with the side-benefit that we may actually respect the "inplace" keyword (see TODO comment L1508)

Reply:
Well, the point of the `EA.fillna()` method is that this gets called from the DataFrame/Series fillna, so that an EA can provide a specialized implementation. So I think we should keep that logic.

Reply:
> so that an EA can provide a specialized implementation

does anyone (e.g. pyarrow) actually have a specialized implementation where it makes a real difference?  e.g. is more performant than going through putmask/where?

> the point of the EA.fillna() method is that this gets called from the DataFrame/Series fillna

I see "the point" of the EA interface in general as being to provide a minimal(ish) set of primitives from which we can  implement the Series/DataFrame(/Index!) API.  If Series.foobar can be implemented efficiently without EA.foobar, that is a reason to remove EA.foobar, not to use it unnecessarily.

(Though in this case we still use EA.fillna for pad/bfill/interpolate cases, so I'm not actually advocating getting rid of it)

Last, while the corrected 'inplace' behavior was not the motivator behind this change, I do think it is an important improvement.